agent/app/agent.go

```diff
@@ -108,6 +110,26 @@ func newAgent(
 		return nil, err
 	}
 
+	var metadataManager containermetadata.MetadataManager
+	if cfg.ContainerMetadataEnabled {
+		// Get the oldest Docker Client version with up to date inspect API
+		// as some metadata is unavailable in older API client versions
+		// If this version is unavailable we use default
+		// Due to limitations of interfaces and import cycles
```

Comment:
I'm not quite sure what you mean here. The problem I am encountering is that adding the WithVersion function to the internal dockerMetadataClient interface in the containermetadata package led to "interface not implemented" errors in the past. I am very uncertain about this section of code as a result, but I will try to see if I can minimize the exposure in this section.